buildtest/buildsystem/base.py

```diff
@@ -527,10 +533,15 @@ def _write_test(self, lines=None):
         return testpath
 
     def _get_test_lines(self):
-        """Given test metadata, get test lines to write to file or show.
-        """
+        """Given test metadata, get test lines to write to file or show."""
+
         # Every test starts with cd to TESTDIR
-        lines = ["cd $TESTDIR"]
+        lines = ["#!/bin/bash"]
+
+        if self.get_shell() == "csh":
+            lines = ["#!/bin/csh"]
```

Comment:
This is what I was saying to avoid - can we always be sure bash and csh are in /usr/bin? We need to find the correct executable before just hard coding one that is an assumption. Take a look at the `executable` variable in the BuildTestCommand to see what I mean. We don't want to hard code something like /bin/csh only to discover that the user doesn't have the binary there.

Reply:
bash and csh are almost always going to be in /bin/csh and /bin/bash. By all means the rpm `tcsh` and `bash` are installed in all systems i have been, without it you can't really do anything. In fact root user is setup with one of the default shell usually bash .

```
(docs) login1.stampede2(1097)$ cat /etc/passwd | grep root
root:x:0:0:root:/root:/bin/bash
```


 Even a bare minimal OS will have these rpms installed.

```
(docs) login1.stampede2(1091)$ rpm -ql tcsh | grep "/bin/*"
/bin/csh
/bin/tcsh

(docs) login1.stampede2(1096)$ rpm -ql bash | grep "/bin/bash"
/usr/bin/bash
/usr/bin/bashbug
/usr/bin/bashbug-64
``` 

Not all systems have tcsh but bash is there all the time. 

I realize that `csh` support can add extra work for buildtest to generate test properly. I have went through this in past, if user doesnt really care then I am all for removing the feature and keeping it simple.

Let me know your thoughts.

Reply:
okay, this is reasonable. We can implement like this until it breaks for someone, and address then I suppose.